lib/routes/cara/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { config } from '@/config';
+import ofetch from '@/utils/ofetch';
+import type { FetchOptions, FetchRequest, ResponseType } from 'ofetch';
+import asyncPool from 'tiny-async-pool';
+import type { PortfolioDetailResponse, PortfolioResponse, UserNextData } from './types';
+import type { DataItem } from '@/types';
+import { parseDate } from '@/utils/parse-date';
+import { API_HOST, CDN_HOST, HOST } from './constant';
+import { load } from 'cheerio';
+
+export function customFetch<T = any, R extends ResponseType = 'json'>(request: FetchRequest, options?: FetchOptions<R>) {
+    return ofetch<T, R>(request, {
+        ...options,
+        headers: {
+            'user-agent': config.trueUA,
+        },
+    });
+}
+
+export async function parseUserData(user: string) {
+    const url = `${HOST}/${user}`;
+    const res = await customFetch(url);
+    const $ = load(res);
+    const nextData = JSON.parse($('#__NEXT_DATA__')?.html() ?? '{}') as UserNextData;
+    return nextData.props.pageProps.user;
+}
```

Comment:
User data can be cached and obtained from the JSON response of `https://cara.app/_next/data/${buildId}/${username}.json?user=${username}`

Regarding `buildId`, it can be obtained from `#__NEXT_DATA__` of a fix page like their homepage `cara.app/explore` and can be cached as well but with a shorter lifespan(i.e., `config.cache.routeExpire`) and `refresh: false` option